orderbook/src/main.rs

```diff
@@ -41,10 +41,12 @@ async fn main() {
     let chain_id = web3.eth().chain_id().await.expect("Could not get chainId");
     let domain_separator =
         DomainSeparator::get_domain_separator(chain_id.as_u64(), settlement_contract.address());
-
     let orderbook = Arc::new(OrderBook::new(domain_separator));
-    let filter = api::handle_all_routes(orderbook.clone())
-        .map(|reply| warp::reply::with_header(reply, "Access-Control-Allow-Origin", "*"));
+    let cors = warp::cors()
+        .allow_any_origin()
+        .allow_methods(vec!["GET", "POST", "DELETE", "OPTIONS"])
+        .allow_headers(vec!["Origin", "Content-Type", "X-Auth-Token", "X-AppId"]);
```

Comment:
Do we want need to be this explicit with the header? 

I'm not expert either, is it more secure than `Access-Control-Allow-Headers: *` ?

Reply:
It doesn't seem to have allow_any_headers, but maybe we can try allow_header("*")

Reply:
Looks like warp doesn't support this currently. Maybe something we could implement for warp.